compiler/rustc_mir/src/borrow_check/diagnostics/mod.rs

```diff
@@ -965,35 +965,33 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
         let expr = &self.infcx.tcx.hir().expect_expr(hir_id).kind;
         debug!("closure_span: hir_id={:?} expr={:?}", hir_id, expr);
         if let hir::ExprKind::Closure(.., body_id, args_span, _) = expr {
-            for (upvar_hir_id, place) in
-                self.infcx.tcx.typeck(def_id.expect_local()).closure_captures[&def_id]
-                    .keys()
-                    .zip(places)
+            for (captured_place, place) in self
+                .infcx
+                .tcx
+                .typeck(def_id.expect_local())
+                .closure_min_captures_flattened(def_id)
+                .zip(places)
             {
-                let span = self.infcx.tcx.upvars_mentioned(local_did)?[upvar_hir_id].span;
+                let span = self.infcx.tcx.hir().span(captured_place.info.expr_id.unwrap());
```

Comment:
Just to reduce the size of this diffs I'd leave this to use upvars_mentioned span with a comment 
`//FIXME(project-rfc-2229#8): Use better span from captured_place`

I'd bring in https://github.com/sexxi-goose/rust/blob/80215733361199d950213f44c7397f6133f11114/compiler/rustc_middle/src/ty/mod.rs#L817-L824 to `middle/src/ty/mod.rs` so and use that here.

Reply:
i dont think it is too big to include this 1 line

Reply:
This results in most of the changes to the test outputs

Reply:
This PR should ideally not change test output that much. That should be a separate PR just to fix diagnostics in borrow checking

Reply:
hmmmmmmm alright